### PR TITLE
replaced `max` and `min` operations for puma function (improves speed)

### PR DESCRIPTION
**Pull request by ChangPaul**

### CODE REVIEW

File: kamui/core.py [hunk 1]
```diff
@@ -185,11 +185,8 @@ def cal_Ek(K, psi, i, j):
 
     prev_Ek = cal_Ek(K, psi, edges[:, 0], edges[:, 1])
 
-    energy_list = []
-
     for step in jump_steps:
         while 1:
-            energy_list.append(prev_Ek)
```
Comment (yoyololicon) [inline kamui/core.py:192]: You're right, the list can be removed.

**Review by yoyololicon — COMMENTED** [review]
Hey @ChangPaul , thanks for the contribution!

Could you provide some benchmarks for your improvements over the previous version? such as runtime.

**Comment by ChangPaul** [comment]
Hey @yoyololicon 

Tests on the wrapped shear (from examples/synthetic_images.ipynb) was previously 1.1119 seconds (average of 50 repetitions) and after the modification was 0.5431 seconds (also average of 50 repetitions). Therefore approximately 2x faster.

Tests on a 3d wrapped shear (`shear = np.repeat(shear[..., None], 20, axis=2)`) was previously 51.1 seconds runtime (for one measurement) and after the modification was 34.4 seconds runtime (for one measurement).

**Review by yoyololicon — APPROVED** [review]
Thanks for reporting the numbers! Great contribution.

**Comment by yoyololicon** [comment]
The improvements will be available in version 0.1.2. 
Thank you @ChangPaul

**Pull request merged**